### ISSUE TITLE
vim test for robot framework

### ISSUE THREAD

**Issue by huynguyen-quoc** [janko/vim-test#456]
Could you please support robot framework for testing

**Comment by codeinabox** [comment]
If you have time could create a pull request with an example of your situation in the fixtures and specs and I'll happily implement the logic for it.

**Comment by huynguyen-quoc** [comment]
@codeinabox  This is the example of robot framework testing i think it's almost similar with cucumber
[https://robotframework.org/#examples](https://robotframework.org/#examples)

**Comment by codeinabox** [comment]
Yeah I saw but I need concrete examples of how it should behave for each thing like TestNearest, TestFile and what CLI commands it should generate as that would make it much easier for me to add support.

**Comment by huynguyen-quoc** [comment]
@codeinabox TestNearest i think it's similar with cucumber 
CLI commands can be like this
> rebot -v env:dev_local -v browser:headlesschrome filename

**Comment by codeinabox** [comment]
> I'm trying to write the runner myself but I'm not a programmer and I'm a bit stuck.
> 
> if I supply a test file and the commands to run the TestNearest en TestFile ect.? can you make a basic runner for robotframework? if that is present I can look to extend the functions.

Happy to lend a hand, could you please create a draft pull request with some fixtures and a spec like those in the `spec` folderm and I can help with the implementation.

**Comment by codeinabox** [comment]
> i made a draft pr with a robotframework fixure and spec file. [#834](https://github.com/vim-test/vim-test/pull/834)

Thank you, this is very useful! I'll have a go at implementing this, sometime this month

**Comment by Samma3ll** [comment]
Don't want to push but do you have any update?
Can I do something to help?

**Comment by codeinabox** [comment]
> Don't want to push but do you have any update? Can I do something to help?

Not forgotten about this, will look into it, and thanks for the nudge

**Comment by Samma3ll** [comment]
another nudge.

**Comment by codeinabox** [comment]
I tried opening a `.robot` file in both vim and neovim, and in both cases there is no file type, which makes things a tad tricky as vim-test needs a file type as part of the detection process.

**Comment by Samma3ll** [comment]
if I open a .robot file en check with `:set ft` i get `filetype=robot` back

![Image](https://github.com/user-attachments/assets/deb16291-cc6b-49e5-a214-76923dbb51c1)

**Comment by codeinabox** [comment]
Initial support for Robot Framework has now been merged in! Could you please try this out, and report any issues here.

**Comment by Samma3ll** [comment]
awesome!
did a quick test and TestFile seems to be working.
TestNearest is also testing the whole file not just the file the cursor is in.

I will test some more this weekend.
but thank you for this.

**Comment by codeinabox** [comment]
Thank you for confirming that `TestFile` is working as expected. I am going to close this issue, and we can create new ones for improvements and bugs.
